src/actions/BreachActions.js

```diff
@@ -23,18 +23,24 @@ const createConsecutiveSuccess = (sensor, updatedBreaches, updatedLogs) => ({
 });
 
 const createConsecutiveBreaches = sensor => async dispatch => {
-  const { id } = sensor;
+  const { id: sensorID } = sensor;
 
   try {
-    const logs = await BreachManager().getLogsToCheck(id);
+    const logs = await BreachManager().getLogsToCheck(sensorID);
+    const mappedToPlainObjects = logs.map(({ id, temperature, timestamp }) => ({
```

Comment:
When we assign a `breach` in the `BreachManager` to a log, we are actually assigning to a realm object which would throw an error because you can only edit realm objects within a realm transaction, so mapped this to a plain object.. this only started happening when I corrected the the logic to actually close off breaches

Reply:
Maybe should add a `toJSON` to the `TemperatureLog` class